charts/karpenter/templates/role.yaml

```diff
@@ -46,6 +46,25 @@ rules:
     resources: ["configmaps"]
     verbs: ["create"]
 ---
+# Karpenter uses this role to discover the Kubernetes APIServer Endpoints
+# This is important for getting consensus on the Kubernetes in the cluster.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "karpenter.fullname" . }}-kubernetes
+  namespace: default
+  labels:
+    {{- include "karpenter.labels" . | nindent 4 }}
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+rules:
+  # Read
+  - apiGroups: ["discovery.k8s.io"]
+    resources: ["endpointslices"]
```

Comment:
This should be scoped.

Reply:
To establish a watch on a resource, it has to be on a collection, so we can't select by name here IIUC. At the moment it's scoped by namespace, and at least in my cluster, I only see kubernetes endpoints here.